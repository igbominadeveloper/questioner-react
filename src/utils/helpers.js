import jwt from 'jsonwebtoken';

export const decodeToken = token => {
  return jwt.decode(token);
};

export const setItem = (name, value) => {
  localStorage.setItem(name, value);
};

export const getItem = name => {
  const item = localStorage.getItem(name);
  return item;
};

export const destroyItem = item => {
  return localStorage.removeItem(item);
};

export const clearLocalStorage = () => {
  return localStorage.clear();
};
