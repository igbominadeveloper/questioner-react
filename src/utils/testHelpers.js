import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);

export const setupStore = initialState => mockStore(initialState);

export const findByTestAttribute = (component, attribute) => {
  const wrapper = component.find(`[data-test='${attribute}']`);
  return wrapper;
};
