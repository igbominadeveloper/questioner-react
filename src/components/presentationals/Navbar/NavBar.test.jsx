import React from 'react';
import { shallow } from 'enzyme';

import NavBar from './Navbar';

import { findByTestAttribute } from '../../../utils/testHelpers';

const component = shallow(<NavBar />);

describe('Custom Search', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders a nav html element', () => {
    const wrapper = findByTestAttribute(component, 'navbar');
    expect(wrapper.length).toBe(1);
  });

  it('renders the navlinks', () => {
    const wrapper = findByTestAttribute(component, 'nav-links');
    expect(wrapper.length).toBe(1);
  });
});
