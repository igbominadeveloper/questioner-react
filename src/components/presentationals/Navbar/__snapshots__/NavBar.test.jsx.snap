// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Search renders without errors 1`] = `
<Fragment>
  <nav
    className="navbar"
    data-test="navbar"
    id="navbar-desktop"
  >
    <div
      className="page-title"
    >
      <Link
        to="/"
      >
        <h3>
          Questioner
        </h3>
      </Link>
    </div>
    <div
      className="nav-links"
      data-test="nav-links"
    >
      <ul>
        <li>
          <Link
            to="#"
          >
            <span
              className="fa fa-meetup"
            >
              Meetups
            </span>
          </Link>
        </li>
        <li>
          <Link
            to="/signup"
          >
            <span
              className="fa fa-user-plus"
            >
              Sign up
            </span>
          </Link>
        </li>
        <li>
          <Link
            to="/login"
          >
            <span
              className="fa fa-user-plus"
            >
              Sign in
            </span>
          </Link>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="bg-primary flex-row space-between pr-1 nav-mobile-parent text-white"
  >
    <a
      className="no-repeat h-0"
      href="#"
      id="nav-toggle"
    />
    <div
      className="m-0 h-0"
    >
      <a
        className="text-white"
        href="index.html"
      >
        <p>
          Questioner
        </p>
      </a>
    </div>
    <div
      className="text-white m-0 h-0"
    >
      <a
        className="text-white"
        href="login.html"
      >
        <p>
          Sign in
        </p>
      </a>
    </div>
  </div>
  <nav
    className="navbar-collapse"
    id="navbar-mobile"
  >
    <div>
      <ul
        className="flex flex-column text-left p-1"
      >
        <li
          className="mt-1"
        >
          <Link
            to="/meetups"
          >
            Meetups
          </Link>
        </li>
        <li
          className="mt-1"
        >
          <Link
            to="/login"
          >
            Login
          </Link>
        </li>
        <li
          className="mt-1"
        >
          <Link
            to="/signup"
          >
            Sign up
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</Fragment>
`;
